lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -2033,6 +2064,7 @@ namespace Js
         SparseArraySegmentBase *seg, *nextSeg, *prevSeg = nullptr;
         for (seg = fArray->head; seg; seg = nextSeg)
         {
+            bool isInlineSegment = JavascriptArray::IsInlineSegment(seg, fArray);
```

Comment:
Point taken and updated in other occurrences. Decided JavascriptNativeFloatArray::ConvertToVarArray() does not need to call clearElements() because its inlined head segment only contains native data. Same thing for JavascriptNativeIntArray::ConvertToVarArray().

---
In reply to: [123652599](https://github.com/Microsoft/ChakraCore/pull/3211#discussion_r123652599) [](ancestors = 123652599)